packages/instantsearch.js/src/widgets/chat/chat.tsx

```diff
@@ -190,6 +197,108 @@ const createRenderer = <THit extends NonNullable<object> = BaseHit>({
       },
     }));
 
+    renderState.templateProps = prepareTemplateProps({
+      defaultTemplates: {} as unknown as NonNullable<
+        Required<ChatTemplates<THit>['prompt']>
+      >,
+      templatesConfig: instantSearchInstance.templatesConfig,
+      templates: templates.prompt,
+    }) as PreparedTemplateProps<ChatTemplates<THit>>;
+    const promptHeaderComponent = templates.prompt?.header
+      ? () => {
+          return (
+            <TemplateComponent
+              {...renderState.templateProps}
+              templateKey="header"
+              rootTagName="fragment"
+            />
+          );
+        }
+      : undefined;
+    const promptFooterComponent = templates.prompt?.footer
+      ? () => {
+          return (
+            <TemplateComponent
+              {...renderState.templateProps}
+              templateKey="footer"
+              rootTagName="fragment"
+            />
+          );
+        }
+      : undefined;
+
+    const promptTranslations: Partial<ChatPromptTranslations> =
+      getDefinedProperties({
+        textareaLabel: templates.prompt?.textareaLabelText,
+        textareaPlaceholder: templates.prompt?.textareaPlaceholderText,
+        emptyMessageTooltip: templates.prompt?.emptyMessageTooltipText,
+        stopResponseTooltip: templates.prompt?.stopResponseTooltipText,
+        sendMessageTooltip: templates.prompt?.sendMessageTooltipText,
+        disclaimer: templates.prompt?.disclaimerText,
+      });
+
+    renderState.templateProps = prepareTemplateProps({
```

Comment:
no it's not expected. You override the values. There should be one templateProps per type

Reply:
Ah right I'll assign it to a new var.

Reply:
Done! I don't think the renderState's template props are going to be used in this case, should we remove it or keep them in case they are needed.